File: GroovyLabSrc/groovySci/math/array/benchMatrix.java

package groovySci.math.array;

import javax.swing.JOptionPane;

public class benchMatrix {

    public static void main(String [] args)
    {
        long start = System.currentTimeMillis();
        int NLoop=1000;
        int N=200;
        
        double [][] m = new double[N][N];
for (int reps=0; reps < NLoop; reps++)
  for (int r=0; r<N; r++)
   for (int c=0; c<N; c++)
      m[r][c] = NLoop*r*c;

long end = System.currentTimeMillis();

double  delay = (double)(end-start)/1000.0;
JOptionPane.showMessageDialog(null, "delay = "+delay);
    }
}
// 0.219 sec 

File: GroovyLab-Src/groovySciExamples/Matrix/benchMatrix.m
tic;
NLoop=1000;
N=200;
m = rand(N, N);

for reps = 1:NLoop,
  for r = 1:N,
   for  c = 1:N,
      m(r,c) = NLoop*r*c;
   end;
  end;
 end;
 
  delay = toc;
  
  disp('delay ='+delay);

  % delay ~ 77.78 sec